openadapt/share.py

```diff
@@ -0,0 +1,119 @@
+utils.configure_logging(logger, LOG_LEVEL)
+
+
+def export_recording_to_folder(recording_id: int) -> None:
+    """Export a recording to a zip file.
+
+    Args:
+        recording_id (int): The ID of the recording to export.
+
+    Returns:
+        str: The path of the created zip file.
+    """
+    recording_db_path = db.export_recording(recording_id)
+
+    if recording_db_path:
+        # Create the directory if it doesn't exist
+        os.makedirs(config.ZIPPED_RECORDING_FOLDER_PATH, exist_ok=True)
+
+        # Path to the source db file
+        db_filename = f"recording_{recording_id}.db"
+
+        # Path to the compressed file
+        zip_filename = f"recording_{recording_id}.zip"
+        zip_path = os.path.join(config.ZIPPED_RECORDING_FOLDER_PATH, zip_filename)
+
+        # Create an in-memory zip file and add the db file
+        with ZipFile(zip_path, "w", ZIP_DEFLATED, compresslevel=9) as zip_file:
+            zip_file.write(recording_db_path, arcname=db_filename)
+
+        logger.info(f"Created zip file of the recording: {zip_path}")
+
+        # delete db file
+        os.remove(recording_db_path)
+        logger.info(f"Deleted db file of the recording: {recording_db_path}")
+
+        return zip_path
+
+
+def send_file(file_path: str) -> None:
+    """Send a file using the 'wormhole' command-line tool.
+
+    Args:
+        file_path (str): The path of the file to send.
+    """
+    # Construct the command
+    command = ["wormhole", "send", file_path]
+
+    # Execute the command
+    try:
+        subprocess.run(command, check=True)
+    except subprocess.CalledProcessError as e:
+        raise Exception(f"Error occurred while running 'wormhole send': {e}")
+
+
+def send_recording(recording_id: int) -> None:
+    """Export a recording to a zip file and send it to another computer.
+
+    Args:
+        recording_id (int): The ID of the recording to send.
+    """
+    zip_file_path = export_recording_to_folder(recording_id)
+
+    if zip_file_path:
+        try:
+            send_file(zip_file_path)
+            # File sent successfully
+        except Exception as e:
+            logger.error(str(e))
+        finally:
+            # Delete the zip file after sending or in case of exception
+            if os.path.exists(zip_file_path):
+                os.remove(zip_file_path)
+                logger.info(f"Deleted zip file of the recording: {zip_file_path}")
+
+
+def receive_recording(wormhole_code: str) -> None:
+    """Receive a recording zip file from another computer using a wormhole code.
+
+    Args:
+        wormhole_code (str): The wormhole code to receive the recording.
+    """
+    # Construct the command
+    command = ["wormhole", "receive", wormhole_code]
+
+    # Execute the command
+    try:
+        subprocess.run(command, check=True)
+    except subprocess.CalledProcessError as e:
+        logger.warning(
+            f"Error occurred while running 'wormhole receive {wormhole_code}': {e}"
+        )
```

Comment:
```logger.exception(exc)```

Reply:
addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/223/commits/3992d1c78d334a352054c2575db248df8677ec4a